File: 0014-longest-common-prefix/0014-longest-common-prefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let prefix = ""
    for(let i = 0; i < strs[0].length; i++){
        for(const str of strs){
            if(i > str.length || strs[0][i] !== str[i]) return prefix
        }
        prefix = prefix.concat(strs[0][i])
    }
    
    return prefix
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs = sorted(strs)
        prefix = ''
        first_word = strs[0]
        last_word = strs[-1]
        for char_idx in range(len(first_word)):
            if(first_word[char_idx] != last_word[char_idx]):
                return prefix
            prefix = prefix + first_word[char_idx]
        
        return prefix